spring-data-mongodb/src/main/java/org/springframework/data/mongodb/gridfs/GridFsTemplate.java

```diff
@@ -247,28 +246,15 @@ public GridFsResource getResource(GridFSFile file) {
      * (non-Javadoc)
      * @see org.springframework.core.io.support.ResourcePatternResolver#getResources(java.lang.String)
      */
-	public GridFsResource[] getResources(String locationPattern) {
-
-		if (!StringUtils.hasText(locationPattern)) {
-			return new GridFsResource[0];
-		}
+    public GridFsResource[] getResources(String locationPattern) {
+        Pattern pattern = Pattern.compile(locationPattern);
 
-		AntPath path = new AntPath(locationPattern);
+        return StreamSupport
+                .stream(find(query(whereFilename().regex(pattern))).spliterator(), false)
+                .map(GridFsResource::new)
```

Comment:
This invocation seems to result in something different than the old constructor invocation. The `getGridFs().…` call is missing. Is this an issue?

Reply:
Yes, you are right, i misused the possibility to instatiate a GridFsResource with a GridFsFile. I thought this would also provide a valid GridFsResource, but as i looked into it seems that the inputStream is missing. I will have a look. Thx for your feedback!

Reply:
Yes, you are right, i misused the possibility to instatiate a GridFsResource with a GridFsFile. I thought this would also provide a valid GridFsResource, but as i looked into it seems that the inputStream is missing. I will have a look. Thx for your feedback!